src/main/webapp/dev/js/common/questionMsq.js

```diff
@@ -83,23 +60,19 @@ function adjustMaxSelectableChoices(questionNum) {
 
     const upperLimit = isGenerateOptionsEnabled(questionNum)
             ? getTotalOptionsForSelectedGenerateOptionsType(questionNum) : getNumOfMsqOptions(questionNum);
-    const currentVal = getMaxSelectableChoicesValue(questionNum);
 
     setUpperLimitForMaxSelectableChoices(questionNum, upperLimit);
-    setMaxSelectableChoices(questionNum, Math.min(currentVal, upperLimit));
```

Comment:
@wkurniawan07 This function changes the current value to upper limit if it is greater than upper limit without notifying user. After removing it, user can enter any value and when he tries to submit, if the current value is greater than upper limit then submit action is blocked and notification is given. Is my approach to this issue incorrect?

Reply:
@nidhi98gupta That is not the problem. Any code block must be present in the code base for a reason. What you're doing right now is analogous to "this test is failing and I don't know how to fix it; let's just remove it", which I am really wary of. Are you really sure that this removed code block did not do whatever it was supposed to, and thus needs to be removed?

Reply:
If what we are looking for is just a html form alert then I think adding it to the template should suffice?
https://github.com/TEAMMATES/teammates/blob/5d2a02cf6b0cba6279d2935b0e7877294fd3d2b4/src/main/resources/feedbackQuestionMsqEditFormTemplate.html#L62-L67

Reply:
@bqnguyen94 Are you suggesting using built-in alert or customized one? Can you please elaborate.
@wkurniawan07 There are two submit buttons in question form. Top one gives customized alert message and saves the form after changing current value to upper bound. Bottom one doesn't.
![uparkierror](https://user-images.githubusercontent.com/24381442/36389838-a827632a-15c6-11e8-99dc-3de6bef072fb.png)
Should I change the behavior of bottom submit button similar to the top one?

Reply:
![image](https://user-images.githubusercontent.com/11720085/36404560-4f102f64-1626-11e8-84d0-f85eecd82e72.png)
The notification in this screenshot is built-in. It does not appear for maximum value because the `input` doesn't have a `max`. Anyway, I think there should be 2 notifications, the form alert and another alert displayed when an invalid form data was submitted, instead of just replacing the invalid value and showing no alert to user.

Reply:
@bqnguyen94 I think `input` does have `max`. I checked it with inspector. And to be sure we are setting it from https://github.com/TEAMMATES/teammates/blob/03917c8c2d250ff2fe84726e7ae66354169e53d4/src/main/webapp/dev/js/common/questionMsq.js#L35
The reason built in notification does not appear is, it is getting reset to upper limit  with this code `setMaxSelectableChoices(questionNum, Math.min(currentVal, upperLimit));`

Reply:
@wkurniawan07 Requesting your inputs to move ahead in this PR.

Reply:
Ah I see. Thanks for the clarification. Then the problem is probably here:
https://github.com/TEAMMATES/teammates/blob/03917c8c2d250ff2fe84726e7ae66354169e53d4/src/main/webapp/dev/js/common/questionMsq.js#L213-L221
Looks like is deliberate that value of number input field is adjusted upon change and no warning shown is expected?

Reply:
@bqnguyen94 If we further inspect this function https://github.com/TEAMMATES/teammates/blob/03917c8c2d250ff2fe84726e7ae66354169e53d4/src/main/webapp/dev/js/common/questionMsq.js#L79-L90
we can probably zero the problem to line number 89?